scripts/generators/rules_generator.py

```diff
@@ -0,0 +1,81 @@
+import json
+import os.path
+import sys
+
+from generator import ecs_helpers
+
+
+def generate(ecs_flat, version, out_dir):
+    rules = []
+    rules.extend(gen_required_rules(ecs_flat))
+    rules.extend(gen_allowed_values_rules(ecs_flat))
+    rules.extend(gen_category_expected_types_rules(ecs_flat))
+    if len(rules) > 0:
+        ecs_helpers.make_dirs(os.path.join(out_dir, 'rules'))
+        save_rules(os.path.join(out_dir, 'rules/rules.json'), rules)
+
+
+def gen_required_rules(ecs_flat):
+    rules = []
+    required_fields = dict(filter(lambda elem: 'required' in elem[1], ecs_flat.items()))
+    for field_name, field_properties in required_fields.items():
+        if field_properties['required'] == False:
+            next
+        rule = {"name": "ECS Check: {0} required field".format(field_name),
```

Comment:
A possible change for later:

Similar how some of the generated asciidoc files use [templates](https://github.com/elastic/ecs/tree/master/scripts/templates), we could consider creating rule templates to have some of the rule configuration outside the rule generator here.